pkg/kubelet/cm/devicemanager/topology_hints_test.go

```diff
@@ -316,6 +316,39 @@ func TestGetTopologyHints(t *testing.T) {
 				},
 			},
 		},
+		{
+			description:   "Single device type, some allocated to container",
+			podUID:        "fakePod",
+			containerName: "fakeContainer",
+			request: map[string]string{
+				"testdevice": "1",
+			},
+			devices: map[string][]pluginapi.Device{
+				"testdevice": {
```

Comment:
To test what we are trying to verify here, there shouldn't be devices in this list that match devices that are already allocated to the container.

Reply:
For the following setup:
```
			devices: map[string][]pluginapi.Device{
				"testdevice": {
					makeNUMADevice("Dev2", 1),
				},
			},
```
There is no hint returned.

Reply:
Yes, but the point of this patch is to make sure that we get a hint back for the "testdevice" resources already allocated to the container, even if there are no devices of that type registered with the kubelet. I.e. No devices of type "testdevice" in the `devices` list at all.

Reply:
Stepped through ManagerImpl#generateDeviceTopologyHints, the following condition was true for every iteration:
```
			if m.allDevices[resource][d].Topology == nil {
				continue
```
which led to the following return (line 137):
```
		if numMatching < request {
			return
```

Reply:
Right. So it looks like this simple patch isn't sufficient to resolve the issue then, unfortunately. Will need to think about the best way to address it.

Reply:
The issue is that we don't actually checkpoint the NUMA information for the devices anywhere. So if the plugin hasn't reregistered yet, then there is actually no way to generate the hints at this point unfortunately.